packages/react/src/internal/useEvent.js

```diff
@@ -21,10 +21,33 @@ export function useEvent(element, eventName, callback) {
       }
     }
 
+    const element = elementOrRef.current ?? elementOrRef;
     element.addEventListener(eventName, handler);
 
     return () => {
       element.removeEventListener(eventName, handler);
     };
-  }, [element, eventName]);
+  }, [elementOrRef, eventName]);
+}
+
+export function useWindowEvent(eventName, callback) {
```

Comment:
This will have differing behavior between React 17 and 18. State updates originating from native event handlers in 17 are not automatically batched, but they will be in 18. https://github.com/reactwg/react-18/discussions/21

It shouldn't be a problem, but wanted to mention it.

Reply:
Thanks for the heads up!

> This shouldn't be necessary because we handle discrete events originating outside of the React event system the same way, by reading window.event. There may be some use cases that don't automatically work though, and for those you can use flushSync. It should be used sparingly though, because if it's done too much you'll just effectively force React back into sync mode.

Found this comment in there, seems like we're okay to move forward but need to be aware of this / add it in for specific situations?

Reply:
@joshblack Yeah exactly 👍